app.py

```diff
@@ -107,14 +207,132 @@ def do_logout(request):
 
 @view_config(route_name="question", renderer='templates/questionpage.jinja2')
 def question(request):
-    return {}
+    if request.authenticated_userid:
+        user = User.get_by_username(request.authenticated_userid)
+        if request.method == "POST":
+            answer = request.params.get("answer")
+            question = Question.get_question_by_id(
+                request.params.get("question_id")
+            )
+            if (answer not in [None, ''] and question.id not in
+               [sub.question_id for sub in Submission.get_all_for_user(user)]):
+
+                Submission.new(
+                    user=user,
+                    question=question,
+                    answer=answer
+                )
+        questions = Question.all()
+        if questions:
+            submissions = Submission.get_all_for_user(user)
+            l = []
+            for q in questions:
+                if q.id not in [s.question_id for s in submissions]:
+                    l.append(q)
+            if l:
+                question = l[randint(0, len(l) - 1)]
+                x, u, y = make_data(question, user)
+                if x != [] and y != []:
+                    prediction = int(round(guess(x, u, y)))
+                else:
+                    prediction = "Not enough data to predict this question. Keep going!"
+                if 'HTTP_X_REQUESTED_WITH' in request.environ and request.method == "POST":
+                    return Response(
+                        body=json.dumps({"text": question.text,
+                                         "qid": question.id,
+                                         "prediction": prediction,
+                                         }), content_type=b'application/json')
+                return {"question": question, "prediction": prediction}
+        return {"question": None, "prediction": None}
+    else:
+        return HTTPFound(request.route_url('home'))
 
 
-@view_config(route_name="faq", renderer='templates/faqpage.jinja2')
+@view_config(route_name="about", renderer='templates/faqpage.jinja2')
 def faq(request):
     return {}
 
 
+def make_data(question, user):
+    """Gets data from the database and parses it for the Guess function"""
+    u, questions = _get_data(user, question)
+    users, questions = _select_users(u, questions)
+    x = _parse_into_matrix(questions, users)
+    y = []
+    for user_ in users:
+        y.append(Submission.get_answer(user_, question))
+    u = [sub.answer for sub in Submission.get_all_for_user(user)]
+    u = [Submission.get_answer(user, q) for q in questions]
+    return x, u, y
```

Comment:
I'm finding the path through these functions difficult enough to grok that I'd really like there to be better docstrings in here.  I also feel like if there were good docstrings for these function, it might help you to see more efficient ways to reach the same goals.